tasks/gov.js

```diff
@@ -37,31 +46,247 @@ task("gov:setPricingConfig")
+                let healthScore = liabilityValue == 0 ? ethers.constants.MaxUint256 : (collateralValue * et.c1e18) / liabilityValue;
+
+                health_scores[account] = {
+                    health: healthScore / et.c1e18,
+                    collateralValue,
+                    liabilityValue
+                };
+            }));
+            console.log("*** The batch has been resolved")
+        }
+
+        let outputJson = JSON.stringify(health_scores);
+        fs.writeFileSync(`${filename}.json`, outputJson + "\n");
+    });
+
+
+
+task("gov:forkHealthScoreDiff", "Compare the health scores of accounts from a pair of JSON files")
+    .addPositionalParam("preGovActionFileName", "file name without .json suffix")
+    .addPositionalParam("postGovActionFileName", "file name without .json suffix")
+    .addPositionalParam("removeFilesAfterParsing", "true or false")
+    .setAction(async ({ preGovActionFileName, postGovActionFileName, removeFilesAfterParsing }) => {
+        const fs = require('fs');
+
+        const prePath = `${preGovActionFileName}.json`;
+        const postPath = `${postGovActionFileName}.json`;
+
+        try {
+            const pre_gov_scores = require(`../${prePath}`);
+            const post_gov_scores = require(`../${postPath}`);
+
+            for (let account of Object.keys(pre_gov_scores)) {
+                let collateralValueBefore = ethers.utils.formatEther(pre_gov_scores[account].collateralValue.hex);
+                let liabilityValueBefore = ethers.utils.formatEther(pre_gov_scores[account].liabilityValue.hex);
+                let collateralValueAfter = ethers.utils.formatEther(post_gov_scores[account].collateralValue.hex);
+                let liabilityValueAfter = ethers.utils.formatEther(post_gov_scores[account].liabilityValue.hex);
+                let spyModeURL = `https://app.euler.finance/account/0?spy=${account}`;
+                
+                let result = {
+                    spyModeURL,
+                    healthScoreBefore: pre_gov_scores[account].health,
+                    healthScoreAfter: post_gov_scores[account].health,
+                    collateralValueBefore,
+                    collateralValueAfter,
+                    liabilityValueBefore,
+                    liabilityValueAfter
+                }
+                if (
+                    pre_gov_scores[account].health > 1.15 &&
+                    post_gov_scores[account].health >= 1 &&
+                    post_gov_scores[account].health <= 1.15
+                ) {
+                    console.log(`Account ${account} is at risk of liquidation due to governance action`);
+                    console.log(result);
+                } else if (
+                    pre_gov_scores[account].health > 1 &&
+                    post_gov_scores[account].health < 1
+                ) {
+                    console.log(`Account ${account} is in violation due to governance action`);
```

Comment:
maybe here it'd be good to call the `checkLiquidation` function to calculate how much loss for the violator would be incurred? needs to be double checked, but it seems that the value lost would be: `yield + repay * 0.02` from the `LiquidationOpportunity` struct returned

Reply:
however, it might be a bit tricky as collateral and underlying addresses would be needed as an input

Reply:
That's right. We will need the collateral and underlying addresses. I suppose we can get the report from here then simulate liquidations on the mainnet fork with modified versions of the hardhat tasks in `tasks/liquidate.js`.